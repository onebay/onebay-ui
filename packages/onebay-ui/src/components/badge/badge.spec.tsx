import { mount } from '@vue/test-utils'
import Badge from './index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(Badge, {
    slots,
    components: {},
    props: { ...values }
  })
}

describe('Badge Snap', () => {
  it('render initial Badge', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Loading -- props dot', () => {
    const wrapper = factory({
      dot: true
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Badge -- props maxValue', () => {
    const wrapper = factory({ value: '10', maxValue: 9 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Badge -- props value less than maxValue', () => {
    const wrapper = factory({ value: '4', maxValue: 9 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Badge -- props value not valid', () => {
    const wrapper = factory({ value: 's', maxValue: 9 })
    expect(wrapper.element).toMatchSnapshot()
  })
})
