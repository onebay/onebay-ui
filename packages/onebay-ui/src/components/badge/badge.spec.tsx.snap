// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Snap render Badge -- props maxValue 1`] = `
<div
  class="ob-badge"
>
  
  
  <!---->
  <div>
    <div
      class="ob-badge__num"
    >
      9+
    </div>
  </div>
</div>
`;

exports[`Badge Snap render Badge -- props value less than maxValue 1`] = `
<div
  class="ob-badge"
>
  
  
  <!---->
  <div>
    <div
      class="ob-badge__num"
    >
      4
    </div>
  </div>
</div>
`;

exports[`Badge Snap render Badge -- props value not valid 1`] = `
<div
  class="ob-badge"
>
  
  
  <!---->
  <div>
    <div
      class="ob-badge__num"
    >
      s
    </div>
  </div>
</div>
`;

exports[`Badge Snap render Loading -- props dot 1`] = `
<div
  class="ob-badge"
>
  
  
  <div
    class="ob-badge__dot"
  />
  <!---->
</div>
`;

exports[`Badge Snap render initial Badge 1`] = `
<div
  class="ob-badge"
>
  
  
  <!---->
  
</div>
`;
